zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1514,21 +1514,22 @@ public NotebookEventListener getNotebookInformationListener() {
   }
 
   private void sendAllAngularObjects(Note note, NotebookSocket conn) throws IOException {
+    String user = SecurityUtils.getPrincipal();
```

Comment:
isn't it better to pass `fromMessage->principal` from upper level caller function as a `user`?  it would be more consistent with workflow and I'm not sure whether `SecurityUtils.getPrincipal()` works as expected in `NotebookServer`